File: C#/ShellSort.cs
using System;

class ShellSort
{
    public static void Sort(int[] arr)
    {
        int n = arr.Length;
        for (int gap = n / 2; gap > 0; gap /= 2)
        {
            for (int i = gap; i < n; i++)
            {
                int temp = arr[i];
                int j;
                for (j = i; j >= gap && arr[j - gap] > temp; j -= gap)
                    arr[j] = arr[j - gap];

                arr[j] = temp;
            }
        }
    }

    public static void Main()
    {
        int[] arr = { 12, 34, 54, 2, 3 };
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}

File: Kotlin/shellSort.kt
fun shellSort(arr: IntArray) {
    val n = arr.size
    var gap = n / 2

    while (gap > 0) {
        for (i in gap until n) {
            val temp = arr[i]
            var j = i

            while (j >= gap && arr[j - gap] > temp) {
                arr[j] = arr[j - gap]
                j -= gap
            }
            arr[j] = temp
        }
        gap /= 2
    }
}

fun main() {
    val array = intArrayOf(12, 34, 54, -5, 2, -1, 3, 0)
    shellSort(array)
    println("Sorted Array: ${array.joinToString(", ")}")
}